paymentsheet/src/main/java/com/stripe/android/common/coroutines/CoalescingOrchestrator.kt

```diff
@@ -0,0 +1,80 @@
+package com.stripe.android.common.coroutines
+
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.CoroutineStart
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.async
+import kotlinx.coroutines.coroutineScope
+
+/**
+ * This class intends to simplify workflows where calling a factory multiple times in parallel only results in a single
+ * call to the underlying factory, but the data returned is shared between all callers.
+ */
+internal class CoalescingOrchestrator<T : Any>(
+    private val factory: suspend () -> T,
+    private val keepDataInMemory: (T) -> Boolean = { false },
```

Comment:
Any plans to use this somewhere? No need to remove it if not because it is useful but curious.

Reply:
No plans right now. I wrote this code a long time ago in https://github.com/okta/okta-mobile-kotlin/blob/master/auth-foundation/src/main/java/com/okta/authfoundation/util/CoalescingOrchestrator.kt

So I mostly just took it verbatim (with the tests).